build-infrastructure/README.md

```diff
@@ -0,0 +1,56 @@
+# Building the ECS Agent using AWS CodeSuite
+
+## Prerequisites
+
+There are some things you need to get this going
+
+1. Your favorite AWS region
+1. A GPG key stored in AWS Secrets Manager
+1. A passphrase for the aforementioned GPG key, also stored on AWS Secrets Manager
+1. A codestar github connection
+1. A final artifacts bucket
+
+## How to get the key and the connection
+
+Generate and export a key for yourself by running the following commands.
+
+```shell
+gpg --full-generate-key
+gpg --output private.gpg --armor --export-secret-key <email>
+```
+
+A couple of things to keep in mind. If you're using Amazon Linux, it comes with a slightly older `gpg` executable so it doesn't _yet_ support ECC keys; best bet is to pick RSA 4096-bit. You will also need a passphrase for the key because it's better security.
+
+You will then have to import this key into secrets manager. To set expectations, plaintext in the context of creating secrets means unstructured (non JSON formatted) data, rather than unencrypted data. Be sure to pick plaintext and just paste the contents of the key into the textbox without JSON encoding it. Secrets Manager encourages the use of JSON structured data but it is not required. You will also need to create a Secrets Manager secret for the passphrase the same way as above.
+
+As for the codestar connection, you can generate one of those by logging into the AWS Console, going to any of the CodeSuite services, clicking on the Settings left nav item and clicking connections. And then clicking the new button. You'll need the ARN of that connection.
+
+## Release pipeline
+
+Grab the `release-pipeline-stack.yml` file and either upload it to CloudFormation or paste it in the CloudFormation designer and create a stack. The stack will ask you for some of the information that you've created above and will generate the following resources
+
+1. A CloudWatch Logs group for the whole CodePipeline
```

Comment:
Is it preferable to keep the logs in CW? Can we get the logs in build directory referenced with commit sha as well?

Reply:
The benefit of the cloudwatch logs is that we can run insights on it and we can search logs using a particular commit id and it becomes easier to have all the logs together!